### PR TITLE
config doc generator  top level grouping refactor

### PR DESCRIPTION
**Pull request by machi1990**
Follow up https://github.com/quarkusio/quarkus/pull/8155 and most precisely this comment https://github.com/quarkusio/quarkus/pull/8155#discussion_r398112922

/cc @geoand thanks for the initial work and  sorry, I did not have the time to check it yesterday, but here it is. Let me know what you think.

### CODE REVIEW

File: core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java [hunk 1]
```diff
@@ -67,12 +69,13 @@ ScannedConfigDocsItemHolder findInMemoryConfigurationItems() {
...
      * @param element - root element
-     * @param parentName - root name
+     * @param rootName - root name
+     * @param parentName - parent name
      * @param configPhase - configuration phase see {@link ConfigPhase}
      * @param withinAMap - indicates if a a key is within a map or is a map configuration key
      * @param sectionLevel - section sectionLevel
      */
-    private List<ConfigDocItem> recursivelyFindConfigItems(Element element, String parentName,
+    private List<ConfigDocItem> recursivelyFindConfigItems(Element element, String rootName, String parentName,
```
Comment (geoand) [inline core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java]: When would `rootName` be different than `parentName`?

**Comment by geoand** [comment]
Thanks a lot of the follow up @machi1990! 

I've added a comment.

File: core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java (hunk 1 above)
Comment (machi1990) [reply core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java]: When scanning group configs items -see  https://github.com/quarkusio/quarkus/pull/8183/files#diff-5b3aaa451926ba3ee4ee4207f00433f1R289

So taking the `liveness-probe` config group for example, we'll have `kubernetes` as root name,  and `kubernetes.liveness-probe` as parent name.

Comment (machi1990) [reply core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java]: That's to say, the `rootName` does not change with depth while the `parentName` changes.

Comment (geoand) [reply core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java]: Gotcha, thanks!

**Review by geoand — APPROVED** [review]
Thanks a lot!

**Pull request merged**